tests/MicrodataParserTest.php

```diff
@@ -102,7 +112,12 @@ public function data()
         ];
     }
 
-    private function getTestData($folderName, $sourceName, $resultName)
+    #[ArrayShape([
+        'uri' => "mixed|string",
+        'source' => "false|string",
+        'result' => "false|string"
+    ])]
```

Comment:
Let's use the DocBlock version as it feels more natural and lean, and is supported by many tools like phpstan, phan, psalm, etc. which means the community is agreed on this solution. [PHPStorm supports this syntax too](https://blog.jetbrains.com/phpstorm/2021/07/phpstorm-2021-2-release/#array_shapes_phpdoc_syntax_support).
The VSCode extension doesn't support it yet, but it will be a bonus when it gets supported: https://github.com/bmewburn/vscode-intelephense/issues/249

```suggestion
    /**
     * @return array{uri: string, source: false|string, result: false|string}
     */
```